### PR TITLE
Collection management UI

### PR DESCRIPTION
**Pull request by niranjan-ramesh**
Addresses #408 . Keeping the issue open to figure out how the conditions for each column of the dataset has to accepted from users. Currently not shown in the form and sends an empty list ```[]```

### CODE REVIEW

**Review by Stephen-ONeil — CHANGES_REQUESTED** [review]
Partial review

File: ui/src/schema/utils.ts [hunk 1]
```diff
@@ -65,3 +65,41 @@ export const constructErrorMessage = (error: ErrorObject): string => {
...
+  name_fr: string;
+  description_en: string;
+  description_fr: string;
+  data_type: string;
+  conditions: {
+    condition_type: string;
+    parameters: string[];
+  }[];
+}
+export interface Collection {
```
[the remark this answers is not in the document]
Comment (niranjan-ramesh) [reply ui/src/schema/utils.ts]: Addressed in [63cd011](https://github.com/HC-IPPM/safe-inputs/commit/63cd0113843eb495fcb8d9c737b9610fb1847a7b). Except for the graphql queries and types in the old stuff.

File: ui/src/components/CollectionForm.tsx [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    update_collection(
+      collection_id: $collection_id
+      collection_updates: $collection_updates
+    ) {
+      id
+    }
+  }
+`;
+
+function renderEmailFields({
```
[the remark this answers is not in the document]
Comment (niranjan-ramesh) [reply ui/src/components/CollectionForm.tsx]: Added separate component for EmailFields in [06b2ef1](https://github.com/HC-IPPM/safe-inputs/pull/578/commits/06b2ef1174ceee30c2636a3d70dc65be0204c41a)[06b2ef1](https://github.com/HC-IPPM/safe-inputs/pull/578/commits/06b2ef1174ceee30c2636a3d70dc65be0204c41a)

File: ui/src/components/CollectionForm.tsx [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+                type="email"
+                placeholder="Enter email"
+                {...field}
+                isRequired
+              />
+            )}
+          />
+          <Button colorScheme="red" onClick={() => removeEmail(index)}>
+            <Trans>Remove</Trans>
+          </Button>
```
[the remark this answers is not in the document]
Comment (niranjan-ramesh) [reply ui/src/components/CollectionForm.tsx]: Refactored with ```aria-label``` in [94ca549](https://github.com/HC-IPPM/safe-inputs/commit/94ca54934142fe264df4067358e560cf6aafab9f)

[the remark this answers is not in the document]
Comment (Stephen-ONeil) [reply ui/src/components/CollectionForm.tsx]: Remember to import and add the lingui template tag `t` to text strings (outside of `<trans>` tags), so lingui can pick them up!

File: ui/src/components/CollectionForm.tsx [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+          emailFields: uploaderEmailFields,
+          control,
+          fieldName: 'uploaders',
+          appendEmail: appendUploaderEmail,
+          removeEmail: removeUploaderEmail,
+        })}
+
+        <Button type="submit" colorScheme="teal" isLoading={loading}>
+          <Trans>Update Collection Details</Trans>
+        </Button>
```
[the remark this answers is not in the document]
Comment (niranjan-ramesh) [reply ui/src/components/CollectionForm.tsx]: Chakra's [```Button```](https://v2.chakra-ui.com/docs/components/button/props#:~:text=string%20%7C%20JSXElementConstructor%3Cany%3E%3E-,loadingText,-Description) has a ```loadingText``` prop that renders a span inside the button when ```loading``` is true. I've set the prop in [f118b77](https://github.com/HC-IPPM/safe-inputs/pull/578/commits/f118b77ad90e2901b1dd154a770c22f4c05e4b5f).

File: ui/src/components/CollectionForm.tsx [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+        });
+        navigate(`/manage-collection/${id}`, {
+          state: {
+            redirect: true,
+          },
+        });
+      }
+    } catch (e) {
+      console.error('Error updating collection:', e);
+    }
```
[the remark this answers is not in the document]
Comment (niranjan-ramesh) [reply ui/src/components/CollectionForm.tsx]: The mutation state does indeed capture the error when its fired. I've removed the ```try```/```catch``` in [f118b77](https://github.com/HC-IPPM/safe-inputs/pull/578/commits/0c7b0c83434cf2e3b41c6d1aefed045bc4649dae).

**Pull request merged**